sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs

```diff
@@ -0,0 +1,328 @@
+    ///   via containment and delegate operations to it.
+    /// </summary>
+    ///
+    /// <seealso cref="Azure.Messaging.ServiceBus.Core.TransportSender" />
+    ///
+    internal class AmqpSender : TransportSender
+    {
+        /// <summary>Indicates whether or not this instance has been closed.</summary>
+        private bool _closed = false;
+
+        /// <summary>The count of send operations performed by this instance; this is used to tag deliveries for the AMQP link.</summary>
+        private int _deliveryCount = 0;
+
+        /// <summary>
+        ///   Indicates whether or not this producer has been closed.
+        /// </summary>
+        ///
+        /// <value>
+        ///   <c>true</c> if the producer is closed; otherwise, <c>false</c>.
+        /// </value>
+        ///
+        public override bool IsClosed => _closed;
+
+        /// <summary>
+        ///   The name of the Service Bus entity to which the producer is bound.
+        /// </summary>
+        ///
+        private string EntityName { get; }
+
+
+        /// <summary>
+        ///   The policy to use for determining retry behavior for when an operation fails.
+        /// </summary>
+        ///
+        private ServiceBusRetryPolicy RetryPolicy { get; }
+
+        /// <summary>
+        ///   The AMQP connection scope responsible for managing transport constructs for this instance.
+        /// </summary>
+        ///
+        private AmqpConnectionScope ConnectionScope { get; }
+
+        /// <summary>
+        ///   The maximum size of an AMQP message allowed by the associated
+        ///   producer link.
+        /// </summary>
+        ///
+        /// <value>The maximum message size, in bytes.</value>
+        ///
+        private long? MaximumMessageSize { get; set; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="AmqpSender"/> class.
+        /// </summary>
+        ///
+        /// <param name="entityName">The name of the entity to which messages will be sent.</param>
+        /// <param name="connectionScope">The AMQP connection context for operations.</param>
+        /// <param name="retryPolicy">The retry policy to consider when an operation fails.</param>
+        ///
+        /// <remarks>
+        ///   As an internal type, this class performs only basic sanity checks against its arguments.  It
+        ///   is assumed that callers are trusted and have performed deep validation.
+        ///
+        ///   Any parameters passed are assumed to be owned by this instance and safe to mutate or dispose;
+        ///   creation of clones or otherwise protecting the parameters is assumed to be the purview of the
+        ///   caller.
+        /// </remarks>
+        ///
+        public AmqpSender(string entityName,
+                            AmqpConnectionScope connectionScope,
+                            ServiceBusRetryPolicy retryPolicy)
```

Comment:
nit: Indentation is off.

Reply:
Fixed pending decision on how to indent these.